src/ui/MLFacet/MLFacetHeader/MLFacetHeader.ts

```diff
@@ -10,18 +10,19 @@ import { MLFacetHeaderCollapseToggle } from './MLFacetHeaderCollapseToggle';
 export class MLFacetHeader {
   public static showLoadingDelay = 2000;
   public element: HTMLElement;
+  private title: Dom;
   private waitAnimation: Dom;
   private clearButton: MLFacetHeaderButton;
   private collapseToggle: MLFacetHeaderCollapseToggle;
   private showLoadingTimeout: number;
 
   constructor(private facet: MLFacet) {
     this.element = $$('div', { className: 'coveo-ml-facet-header' }).el;
-    const titleElement = this.createTitle();
-    $$(this.element).append(titleElement);
+    this.title = this.createTitle();
+    $$(this.element).append(this.title.el);
     $$(this.element).append(this.createWaitAnimation());
     $$(this.element).append(this.createClearButton());
-    this.facet.options.enableCollapse && $$(this.element).append(this.createCollapseToggle());
+    this.facet.options.enableCollapse && this.enableCollapse();
```

Comment:
There should be a top level function on the ML facet component available, to allow to expand and collapse.

This is so that clients that want to control this behaviour don't have to use the dirty trick of targeting the css class and simulating a "click" on the element.

Another way to look at it: it should be possible to test or play with the functionality in it's entirety in your unit tests, without having to always simulate end user click.

If you are able to do that, then it means that if we have clients or PS that want to control stuff in the facet, such as collapse and expand, they will be able to do so easily using the API/programmatically. The UT are now testing for bugs as well as the general "API" of the component.

Testing and simulating end user click is still worthwhile, but should not be the main way you interact with the component in your tests.

Try to think of all the plugins/widget/library you've used in the past.
For example, in google map : 

Of course, the standard way to pan or zoom the map using the mouse. 
But you can be 100% sure that there is a function available in the google map widget to pan/zoom the map using only code, so that people doing integration can create more easily on top of it.

This comment applies to the component as a whole, for all it's functionality.

Reply:
✅